src/main/java/io/appium/java_client/pagefactory/WithTimeout.java

```diff
@@ -38,7 +45,28 @@
     /**
      * Desired time unit.
      *
+     * @deprecated use {@link #chronoUnit()} instead. This property is going to be removed.
      * @return time unit
      */
-    TimeUnit unit();
+    @Deprecated
+    TimeUnit unit() default NANOSECONDS;
+
+    /**
+     * Desired time unit.
+     *
+     * @return time unit. Default value {@link java.time.temporal.ChronoUnit#FOREVER} was added
+     *     for backward compatibility temporary.
+     */
+    ChronoUnit chronoUnit() default FOREVER;
+
+    class DurationBuilder {
+        static Duration build(WithTimeout withTimeout) {
+            //providing backward compatibility
+            if (!FOREVER.equals(withTimeout.chronoUnit())) {
+                return Duration.of(withTimeout.time(), withTimeout.chronoUnit());
+            }
+
+            return Duration.of(MILLISECONDS.convert(withTimeout.time(), withTimeout.unit()), MILLIS);
```

Comment:
is FOREVER the same as milliseconds?

Reply:
I was forced to add defalt values (FOREVER and NANOSECONDS) for bacward compatibility (temporary). Comilation problems will occure otherewise. User will be forced to set both time units.

Reply:
@mykola-mokhnach However
Can we consider FOREVER illegal value? I think it has some sense.

Reply:
I don't know, but it for sure does not help to read and understand the code.